### PR TITLE
add loss/epoch and acc/epoch plots

### PR DESCRIPTION
**Pull request by manonreau**

### CODE REVIEW

File: graphprot/NeuralNet.py [hunk 1]
```diff
@@ -97,25 +97,65 @@ def __init__(self, database, Net,
...
+    def train(self, nepoch=1, validate=False, plot=False):
+        
+        _train_acc = []
+        _train_loss = []
+        _valid_acc = []
+        _valid_loss = []
 
-        self.model.train()
         for epoch in range(1, nepoch+1):
+            self.model.train()
```
Comment (NicoRenaud) [inline graphprot/NeuralNet.py:108]: why do you put train() in the loop  ? Do you change it somewhere in the iteration ?

File: graphprot/NeuralNet.py [hunk 2]
```diff
@@ -97,25 +97,65 @@ def __init__(self, database, Net,
...
+                plt.plot(range(1,nepoch+1), _valid_acc, c='red', label='valid')
+
+            if len(_train_acc) > 1:
+                plt.plot(range(1,nepoch+1), _train_acc, c='blue', label='train')
+                plt.title("Accuracy/ epoch")
+                plt.xlabel("Number of epoch")
+                plt.ylabel("Accuracy")
+                plt.legend()
+                plt.savefig('acc_epoch.png')
+                plt.close()
```
Comment (NicoRenaud) [inline graphprot/NeuralNet.py:158]: So as much as possible we should try to have each function do exactly 1 thing (aka the single responsibility principle). Now if you put the  plot in train(), train() does 2 thing it trains and it plots. As everyone I'm not the best at respecting the single resp ppl but I can try to when I review code :) 

One option would be to have train() return all the losses that can then be used as input of a plot function. Or have member variable for the losses, or export all that in a hdf5 file that you can read afterwards.

**Review by NicoRenaud — CHANGES_REQUESTED** [review]
If you could put the plot part outside of the train() methods that would be great

File: graphprot/NeuralNet.py [hunk 3]
```diff
@@ -95,29 +95,79 @@ def __init__(self, database, Net,
...
+            plt.plot(range(1,nepoch+1), _train_loss, c='blue', label='train')
+            plt.title("Loss/ epoch")
+            plt.xlabel("Number of epoch")
+            plt.ylabel("Total loss")
+            plt.legend()
+            plt.savefig('loss_epoch.png')
+            plt.close()
+            
+            
+    def plot_acc(_train_acc, _valid_acc=[], nepoch):
```
Comment (NicoRenaud) [inline graphprot/NeuralNet.py:117]: I'm surprised it accept a non keyword argument  (nepoch) after a keyword arg (_valid_acc). Here as well I would remove the '_' from the variable name :)

File: graphprot/NeuralNet.py [hunk 4]
```diff
@@ -95,29 +95,79 @@ def __init__(self, database, Net,
 
         elif self.task == 'class':
             self.loss = nn.CrossEntropyLoss(weight = self.class_weights, reduction='mean')       
+        
+        
+    def plot_loss(_train_loss, _valid_loss=[], nepoch):
```
Comment (NicoRenaud) [inline graphprot/NeuralNet.py:100]: I'm getting picky here but usually '_name' variable names are only used as internal. So not used as argument of functions.

**Review by NicoRenaud — COMMENTED** [review]
A few syntax issue that can be easily solved :) Then you can merge and we can see about the hdf5 export later if you want

**Review by NicoRenaud — APPROVED** [review]
